File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int n=0;
        ListNode* temp=head;
        ListNode* temp1=head;
        while(temp!=NULL)
        {
            n++;
            temp=temp->next;
        }
        int c=0;
        while(temp1!=NULL)
        {
            if(c==n/2)
            return temp1;
            else
            c++;
            temp1=temp1->next;
        }
        cout<<c;
        return temp1;
    }
};

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode prev=null;
        ListNode slow=head;
        ListNode fast=head;
        ListNode temp=head;
        int n=0;
        while(temp != null){
            n++;
            temp=temp.next;
        }

        while(fast.next != null || fast != null){
            System.out.println(fast.val);
            prev=slow;
            slow=slow.next;
            if(fast.next == null || fast.next.next == null) break;
            fast=fast.next.next;
        }
        head=slow;
        if(n % 2 != 0)
        head = prev;
        return head;
    }
}